src/sentry/south_migrations/0463_backfill_service_hook_project.py

```diff
@@ -0,0 +1,1344 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
```

Comment:
I think I probably just don't know how DataMigrations work, but why is the `START` at the end and the `COMMIT` at the beginning...?

Reply:
Great question @mnoble. I did this explicitly because the default behavior of a data migration in South is to do the entire migration within a single transaction. This is problematic since we don't want a long live transaction like that. But South is also expecting to commit it's own transaction. So this is a hack to commit the transaction South started, run ours without a transaction, then start a new one so South can commit it. Otherwise, it'll error.

This also makes it easier to surface real errors that may happen within transactions so they don't get masked by the transaction rollback.

Reply:
@mattrobenolt do you have an answer for this? this is just what the template generated and I wasn't sure if I need to add anything